nat-instance.tf

```diff
@@ -67,16 +73,17 @@ data "aws_ami" "nat_instance" {
 # https://docs.aws.amazon.com/vpc/latest/userguide/VPC_NAT_Instance.html
 # https://dzone.com/articles/nat-instance-vs-nat-gateway
 resource "aws_instance" "nat_instance" {
```

Comment:
Could we also change the `aws_instance` to a launch template and ASG ?

Reply:
Technically, yes you could as a NAT instance is started differently than a NAT GW that has its own resource type; however, I don't see the viability of using an ASG to launch it as you need the NAT instance deployed in a specific AZ not floating between AZs assigned to an ASG. Also the EIP association I don't believe is possible with an ASG. If the intention is just to have auto-restart, I believe enabling the CloudWatch metric alarm with a Reboot action would be more appropriate.

Reply:
Yes, See #96 about auto-reboot, and yes, you cannot use EIPs with ASGs as they would have to be dynamically allocated as the Auto Scaling Group scales, which defeats the purpose of having EIPs (which is having stable, consistent IPs over time as resources serving them change).